var/spack/repos/builtin/packages/quantum-espresso/package.py

```diff
@@ -407,6 +409,18 @@ def cmake_args(self):
         if "+qmcpack" in spec:
             cmake_args.append(self.define("QE_ENABLE_PW2QMCPACK", True))
 
+        # Set BLA_VENDOR to All to force detection of vanilla scalapack.
+        # This should be fixed by
+        # https://github.com/QEF/q-e/commit/cf74c944fe8498dd1c423902267cea90ba5d036f
+        if "^armpl-gcc" in spec:
+            cmake_args.append(self.define("BLA_VENDOR", "All"))
+            cmake_args.append(self.define("BLAS_LIBRARIES", spec["armpl-gcc"].libs.joined(";")))
+            cmake_args.append(self.define("LAPACK_LIBRARIES", spec["armpl-gcc"].libs.joined(";")))
+        elif "^acfl" in spec:
+            cmake_args.append(self.define("BLA_VENDOR", "All"))
+            cmake_args.append(self.define("BLAS_LIBRARIES", spec["acfl"].libs.joined(";")))
+            cmake_args.append(self.define("LAPACK_LIBRARIES", spec["acfl"].libs.joined(";")))
+
```

Comment:
Could you describe the issue when addition cmake_args are not added?

Reply:
From what I tested, without adding the pointers to BLAS and LAPACK libraries cmake script is unable to find ArmPL libraries (provided by armpl-gcc or acfl package). The issue with having to set BLA_VENDOR to All is the following: after cmake detects ArmPL libraries it seems to set BLA_VENDOR to Arm. That prevents the cmake script which searches for vanilla scalapack (BLA_VENDOR must be set to either Generic or All).

Reply:
Based on the comment, this issue has been addressed in QE develop and thus the added workaround is only needed for version up-to 7.1. If my understanding is correct, please add a version guard.

Reply:
Version guard to set BLA_VENDOR added. I also merged the two if-blocks together.